src/components/Home/index.js

```diff
@@ -12,18 +12,24 @@ class HomePage extends Component {
     this.state = {
       users: null,
     };
+    this.unsubscribe = null;
   }
 
   componentDidMount() {
-    this.props.firebase.users().on('value', snapshot => {
-      this.setState({
-        users: snapshot.val(),
+    this.unsubscribe = this.props.firebase
+      .users()
+      .onSnapshot(snapshot => {
+        let users = {};
+        snapshot.forEach(doc => (users[doc.id] = doc.data()));
+
+        this.setState({
+          users: users,
+        });
       });
-    });
   }
 
   componentWillUnmount() {
-    this.props.firebase.users().off();
+    this.unsubscribe && this.unsubscribe();
```

Comment:
Interesting. Out of curiosity: Can it actually happen that componentDidMount doesn't define `this.unsubscribe` before will be componentWillMount called? That's why I usually just do:

```
componentDidMount() {
  this.unsubscribe = ...
}

componentWillUnmount() {
  this.unsubscribe();
}
```

without the `this.unsubscribe = null;` in the constructor.

Reply:
Yes, you are right, thank you.